sqlglot/optimizer/eliminate_joins.py

```diff
@@ -87,12 +96,21 @@ def _unique_outputs(scope):
         else:
             return set()
 
-    if all(isinstance(e.unalias(), exp.AggFunc) for e in scope.selects):
+    if _has_single_output_row(scope):
         return set(scope.expression.named_selects)
 
     return set()
 
 
+def _has_single_output_row(scope):
+    return all(isinstance(e.unalias(), exp.AggFunc) for e in scope.selects) or _is_limit_1(scope)
```

Comment:
what about 

Select 1

or 

values ()

Reply:
Oh, woah - I wonder what happens with `VALUES` right now... I don't think the optimizer handles that at all.

Reply:
I'll add logic for a SELECT with no FROM, but I'll skip on VALUES for now